server/plugin.go

```diff
@@ -882,20 +927,22 @@ func (p *Plugin) OnSharedChannelsAttachmentSyncMsg(fi *model.FileInfo, _ *model.
 }
 
 func (p *Plugin) OnSharedChannelsSyncMsg(msg *model.SyncMsg, _ *model.RemoteCluster) (model.SyncResponse, error) {
-	now := model.GetMillis()
-
 	var resp model.SyncResponse
+	if !p.getConfiguration().UseSharedChannels {
+		return resp, nil
```

Comment:
If we don't reply with `resp.PostsLastUpdateAt` set to the incoming value, does that mean all the shared channels are effectively "queued up" until we turn on shared channels? And at that point we'll replay potentially weeks of messages through the plugin and over to MS Teams, possibly risking duplicate messages?

Reply:
Ideally if we go from shared channels off to on, we should set the cursors to current.  That will ensure we don't replay history.  Alternatively we can delete all the SharedChannelRemotes records which contain the cursors when shared channels is turned off, via UninviteRemoteToChannel.

Reply:
Should this something done by the plugin? Looks like a Shared Channels responability. If I disable shared channels during 1 month, and enable it again after that, should it sync the whole month, or since the re-enabling? For me, shared channels, whenever is disabled/enabled should handle that cases.

Reply:
Since this particular configuration is in the plugin, I think this particular decision has to be owned by the plugin. It's perhaps unclear what should happen if the overall shared channels functionality is disabled at the server level.